services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/app/ManagementServicesProducer.java

```diff
@@ -56,6 +57,9 @@ public class ManagementServicesProducer {
     @Inject
     MongoDbBasedCredentialsConfigProperties credentialsServiceProperties;
 
+    @Inject
+    MongoDbDeviceRegistryMetrics metrics;
+
```

Comment:
this property doesn't seem to be used ...

Reply:
If this class is not declared anywhere, the MetricsFactory is not called. I haven't work with Jakarta, is there a way for the MetricsFactory to produce a bean, if there is no references to the bean?

Reply:
Probably not. So, if we nee to keep it then I would suggest to add a comment explaining why we need it ...